### PR TITLE
Introduce user friendlier number parsing system

### PR DESCRIPTION
**Pull request by stuntguy3000**
The new method introduced into StringUtils is the toInteger method which uses regex to remove all invalid numbers from a string, then convert it as required. This is a better change because it allows users to be a little bit more flexible/error resistant to numbers when being inputted.

### CODE REVIEW

File: src/main/java/com/dsh105/nexus/util/StringUtil.java [hunk 1]
```diff
@@ -32,18 +32,17 @@ public static String removePing(String nick) {
...
      *
      * @param string the String to be checked
-     * @return true if Integer
+     * @return Integer.MIN_VALUE if unable to convert
      */
-    public static boolean isInt(String string) {
+    public static int toInteger(String string) {
         try {
-            Integer.parseInt(string);
+            return Integer.parseInt(string.replaceAll("[^\\d]", ""));
```
Comment (lol768) [inline src/main/java/com/dsh105/nexus/util/StringUtil.java]: It's best if you avoid `\d` and go for `0-9` instead since `\d` matches non-English digits.

**Comment by lol768** [comment]
I don't like the whole returning MIN_VALUE for invalid integers to be honest. Why can't we just use exceptions?

File: src/main/java/com/dsh105/nexus/util/StringUtil.java (hunk 1 above)
Comment (stuntguy3000) [inline src/main/java/com/dsh105/nexus/util/StringUtil.java]: Changed.

**Comment by DSH105** [comment]
Instead of returning MIN_VALUE, throw an exception and catch it in the CommandManager.

**Comment by stuntguy3000** [comment]
Taken suggestions above and improved PR.

**Comment by lol768** [comment]
:thumbsup:

**Pull request merged**